boilerplate/openshift/golang-osd-operator/build-opm-catalog.sh

```diff
@@ -0,0 +1,379 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Global vars
+REPO_ROOT=$(git rev-parse --show-toplevel)
+
+## Global vars to modify behaviour of the script
+DRY_RUN=${DRY_RUN:-false}
+DELETE_TEMP_DIR=${DELETE_TEMP_DIR:-true}
+BRANCH=${BRANCH:-master}
+SET_X=${SET_X:-false}
+NO_LOG=${NO_LOG:-false}
+[[ "$SET_X" != "false" ]] && set -x
+
+function log() {
+    local to_log=${1}
+    [[ "$NO_LOG" != "false" ]] && return 0
+    local msg
+    msg="$(date "+%Y-%m-%d %H:%M:%S")"
+    [[ "$DRY_RUN" != "false" ]] && msg="$msg -- DRY-RUN"
+    echo "$msg -- $to_log" 1>&2
+}
+
+function check_required_environment() {
+    local count=0
+    for var in OLM_BUNDLE_IMAGE \
+               OLM_CATALOG_IMAGE \
+               OPERATOR_IMAGE \
+               OPERATOR_IMAGE_TAG \
+               OPERATOR_VERSION \
+               OPERATOR_NAME \
+               CONTAINER_ENGINE \
+               CONTAINER_ENGINE_CONFIG_DIR \
+               CURRENT_COMMIT \
+               OLM_CHANNEL \
+               OLM_BUNDLE_VERSIONS_REPO
+    do
+        if [ ! "${!var:-}" ]; then
+          log "$var is not set"
+          count=$((count + 1))
+        fi
+    done
+
+    [[ $count -eq 0 ]] && return 0 || return 1
+}
+
+function setup_temp_dir() {
+    local temp_dir
+    temp_dir=$(mktemp -d --suffix "-$(basename "$0")")
+    [[ "$DELETE_TEMP_DIR" == "true" ]] && trap 'rm -rf $temp_dir' EXIT
+
+    echo "$temp_dir"
+}
+
+function setup_local_executable() {
+    local executable="${1}"
+    "$REPO_ROOT"/boilerplate/openshift/golang-osd-operator/ensure.sh opm
```

Comment:
1. Use the param.
2. Do you need to make sure this command doesn't print anything to stdout, since you're relying on the output from the function?
```suggestion
    # Redirect to stderr to make sure this output doesn't interfere with stdout expected from the function
    "$REPO_ROOT"/boilerplate/openshift/golang-osd-operator/ensure.sh ${executable} >&2
```

Reply:
1. Yes. That's a bug. It was basically working because I had already everything installed
2. Yes. I wonder how that was working in the first place. I'll fix it

Reply:
2. `ensure.sh` throws everything to stderr, that's why it was working. I'll add your suggestion as it is safer

Reply:
fixed in 3424552